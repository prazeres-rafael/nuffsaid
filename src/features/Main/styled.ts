import styled from "styled-components";
import { Grid } from "@mui/material";

export const Container = styled(Grid)({
  maxWidth: 1440,
  width: "100%",
  margin: "auto",
});

export const Title = styled.h1({
  fontSize: 30,
  fontWeight: 500,
});
