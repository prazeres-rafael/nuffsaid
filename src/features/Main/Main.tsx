import { Grid } from "@mui/material";
import React, { useContext, useEffect, useState } from "react";
import generateMessage, { Message } from "../../Api";
import { Title } from "./styled";
import { ButtonContainer } from "../../components";
import { MessageColumn } from "../index";
import { MessageContext } from "../../context/MessageContext";

export default function Main(): React.ReactElement {
  const { messages, setMessages } = useContext(MessageContext);
  const [getMessages, setGetMessages] = useState<boolean>(true);

  useEffect(() => {
    if (getMessages) {
      const cleanUp = generateMessage((message: Message) => {
        setMessages((oldMessages) => [...oldMessages, message]);
      });
      return cleanUp;
    }
  }, [getMessages]);

  const handleStop = () => {
    setGetMessages(!getMessages);
  };

  const handleClear = () => {
    setMessages([]);
  };

  return (
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <Title>nuffsaid.com Coding Challenge</Title>
        <hr />
      </Grid>
      <Grid item xs={12} sx={{ display: "flex", justifyContent: "center" }}>
        <ButtonContainer
          getMessages={getMessages}
          handleStop={handleStop}
          handleClear={handleClear}
        />
      </Grid>
      <MessageColumn messages={messages} />
    </Grid>
  );
}
