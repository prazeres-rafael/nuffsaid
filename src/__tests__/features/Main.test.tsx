import { render } from "@testing-library/react";
import Main from "../../features/Main/Main";

describe("Main", () => {
  it("should match the snapshot", () => {
    const { container } = render(<Main />);
    expect(container).toMatchSnapshot();
  });
});
