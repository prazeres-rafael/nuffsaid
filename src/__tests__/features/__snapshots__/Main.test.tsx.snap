// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL gSKxLG"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h1
          class="sc-gsDKAQ eOEXXy"
        >
          nuffsaid.com Coding Challenge
        </h1>
        <hr />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-bj7xu4-MuiGrid-root"
      >
        <button
          class="sc-dkPtRN gLjNRF"
        >
          STOP
        </button>
        <button
          class="sc-dkPtRN gLjNRF"
        >
          CLEAR
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <p
          class="sc-iCfMLu ftFgRh"
        >
          Error Type 1
        </p>
        <p>
          Count: 
          0
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <p
          class="sc-iCfMLu ftFgRh"
        >
          Warning Type 2
        </p>
        <p>
          Count: 
          0
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <p
          class="sc-iCfMLu ftFgRh"
        >
          Info Type 3
        </p>
        <p>
          Count: 
          0
        </p>
      </div>
    </div>
  </div>
</div>
`;
